main/src/library/Array.flix

```diff
@@ -335,6 +363,7 @@ namespace Array {
     ///
     /// Reverse the array `a`, mutating it in place.
     ///
+    @Time(length(a)) @Space(1)
     pub def reverse!(a: mut Array[a]): Unit & Impure =
```

Comment:
I like how the annotations illustrate the difference between `reverse` and `reverse!`. Beautiful :)